### PR TITLE
Prefer empty line after an opening brace and beginning of closure. Al…

### PR DESCRIPTION
**Pull request by drewpitchford**
@lgauthier see what you think of these. I think they are good additions. I'm adding cause I intend on getting my team at mSignia to follow this style guide.

### CODE REVIEW

**Comment by drewpitchford** [comment]
adding @skyem to this PR too

**Comment by lgauthier** [comment]
@drewpitchford What's the rationale for the empty line after all opening braces? Personally, I think it would contribute to some unnecessary whitespace that could decrease readability because it would make the use of whitespace more homogeneous throughout the codebase rather than focussed on grouping related code. Either way, if we're going to add it, we'd want to add a rationale after the rule like we do with the other rules.

**Comment by lgauthier** [comment]
@nhgrif Thoughts?

**Comment by drewpitchford** [comment]
@lgauthier I added a rationale. I believe it helps readability by separating the signature from the definition thus making it easier to pick up the beginning of the definition quickly.

**Comment by drewpitchford** [comment]
Any further comments? Or final yay/nay? I did add a section about property declarations too as I'm seeing that is not universally understood either.

**Comment by lgauthier** [comment]
@drewpitchford Sorry for the delay. I've been a little swamped the past couple days. I personally don't like the blank line after opening braces rule, but I'd like to hear what others think. Overall, if there are mixed feelings about it, I'd prefer to make no rule about it so people are free to do it how they prefer. I'd be more open to having a rule stating to always have a blank line after opening braces for function definitions or type definitions, but I like to be able to choose whether to include one or not for control statements and stuff depending on what I think makes it more readable.

I still need to review the other changes proposed here, so I'll get back to you on those hopefully sometime today.

**Comment by lgauthier** [comment]
@drewpitchford Just looked at the property declarations rule. I completely agree that we should have that in there, but I think we can make it even more generic because that's a colon spacing rule that should really apply in many other cases as well. I like this rule from [Ray Wenderlich's style guide](https://github.com/raywenderlich/swift-style-guide):

> Colons always have no space on the left and one space on the right. Exceptions are the ternary operator ? :, empty dictionary [:] and #selector syntax for unnamed parameters (_:).

**Comment by lgauthier** [comment]
Also, we should probably separate these different rule proposals into different PRs.

**Comment by drewpitchford** [comment]
Ok, I'm not sure how to separate them out of this PR other than undoing the work.

**Comment by lgauthier** [comment]
@nhgrif @Truked @BevTheDev You guys should probably weigh in as well before we change anything.

File: README.md [hunk 1]
```diff
@@ -38,22 +40,72 @@ The closing brace for `guard` statements should be followed by an empty line.
...
+
+*Preferred*
+```swift
+override func viewDidLoad() {
+
+    // some code here. Notice the empty line above this one.
+}
+
+// a method with a closure parameter
+myMethod() {
```
Comment (nhgrif) [inline README.md:53]: I believe elsewhere in the style guide, we specify these parentheses should be omitted, so we should fix that.  (If we don't specify it, we should...)  We can just add an argument inside the parenthesis.

File: README.md [hunk 2]
```diff
@@ -364,22 +444,50 @@ It is often best to default to structs, only falling back to classes when you sp
...
+
+    self.view.backgroundColor = .blue
 }
 ```
 
 *Rationale: Omitting `self` allows for more concise code.  Only use it when a local variable has hidden an instance variable or in escaping closures.*
 
 ---
 
+##### Use dot syntax on enums and static vars where type can be inferred.
```
Comment (nhgrif) [inline README.md:465]: Is "static vars" the right phrase for this?  Should these be "static properties"?  And is dropping the type specification referred to as "dot syntax"?  I'm curious if there's a more accurate name for this shortening.

**Review by nhgrif — CHANGES_REQUESTED** [review]
In addition to the comments I've left, I agree with @lgauthier that we should put each section you're adding in its own PR.

**Pull request closed without merging**